tests/test_time.py

```diff
@@ -281,3 +254,46 @@ def test_non_time_dim(self, dummy_calendar, dummy_dataset):
         ds['time'] = np.arange(ds['time'].size)
         with pytest.raises(ValueError):
             dummy_calendar.resample(ds)
+
+class TestTrainTest:
+    # Define all required inputs as fixtures:
+    @pytest.fixture(autouse=True)
+    def dummy_calendar(self):
+        return AdventCalendar(anchor_date=(10, 15), freq="180d")
+
+    def test_set_traintest_method(self, dummy_calendar):
+        dummy_calendar.set_traintest_method("kfold", n_splits = 2)
+        assert dummy_calendar._traintest_method == "kfold" # pylint: disable=protected-access
+        assert dummy_calendar._method_kwargs["n_splits"] == 2 # pylint: disable=protected-access
+
+    def test_set_traintest_method_not_support(self, dummy_calendar):
+        # test when the given method is not supported
+        with pytest.raises(ValueError):
+            dummy_calendar.set_traintest_method("not_a_real_method")
+
+    def test_get_traintest(self, dummy_calendar):
+        dummy_calendar.map_years(2019, 2021, flat=True)
+        dummy_calendar.set_traintest_method("kfold", n_splits = 2)
+        traintest_group = dummy_calendar.get_traintest()
+        # check the first fold
+        expected_group = ['train', 'train', 'test', 'test', 'test', 'test']
```

Comment:
Works because the default option for sklearn is set to `shuffle=False`, otherwise groups would have been random.

Reply:
Yes, I'm aware of it. But maybe not clear for the user/developer. I will add this argument in the function.